src/firebase/auth/auth.ts

```diff
@@ -0,0 +1,120 @@
+import { app } from "../firebase";
+import {
+    query,
+    getDocs,
+    collection,
+    where,
+    addDoc,
+    getFirestore,
+} from "firebase/firestore";
+import { FirebaseError } from "firebase/app";
+
+import {
+    getAuth,
+    GoogleAuthProvider,
+    signInWithPopup,
+    signInWithEmailAndPassword,
+    createUserWithEmailAndPassword,
+    sendPasswordResetEmail,
+    signOut,
+    Auth,
+} from "firebase/auth";
+
+export const db = getFirestore(app);
+export const auth: Auth = getAuth(app);
+
+const googleProvider = new GoogleAuthProvider();
+
+export function getErrorMessage(error: FirebaseError) {
+    switch (error.code) {
+        case "auth/user-not-found":
+            return "Invalid email or password";
+        case "auth/wrong-password":
+            return "Invalid email or password";
+        case "auth/too-many-requests":
+            return "Too many unsuccessful login attempts. Please try again later.";
+        default:
+            return "An error occurred. Please try again.";
+    }
+}
```

Comment:
Remove this please, there will be a separate function in another file/folder to handle this